packages/uniforms/src/ValidatedForm.js

```diff
@@ -33,6 +33,8 @@ const Validated = parent => class extends parent {
         ]).isRequired
     };
 
+    // TODO add `uniforms.state.validating` to `childContextTypes`.
```

Comment:
One more thing.

Reply:
It is not obvious how this should be done. Unlike the actual state, with `getChildContextState`, the static `childContextTypes` is not assembled in this way. What we need is 

```js
static childContextTypes = merge(
  { uniforms: { state: { validating: PropTypes.bool.isRequired } } },
  parent.childContextTypes
);
```

but I don't know of any such `merge` function. Do we need to store the prop types nested in plain objects as well? Then we can have a `shapify` function that turns the plain object tree into a `PropTypes.shapeOf` tree for assignment to `childContextTypes` and the extending classes can merge the plain objects (like `lodash.merge` would) before assigning their own `childContextTypes`.

Or am I missing something simple?

Reply:
There is a way to _merge_ them, but differently - validate it twice. Prop types are just functions, so we can create a new one, which will start with calling the old one.

For now, I'd export an object from the `BaseForm`, marked as temporary (with comment and underscore prefixed name).

Reply:
3640cde is my take on exposing a PropTypes object that can be extended. I'm happy for you to rename and move stuff around, maybe refactor `shapify` and `BaseForm.shapifyPropTypes` into a file or util library.